From c7210c038945277777fd1a59b4d4a6f30db58a9f Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Thu, 20 Feb 2014 15:03:15 +0000
Subject: [PATCH] build GNU make 3.81

If GNU make 3.81 is not found then build that as a companion tool.
---
 config/companion_tools.in | 11 +++++++++++
 configure.ac              |  6 ++++++
 2 files changed, 17 insertions(+)

diff --git a/config/companion_tools.in b/config/companion_tools.in
index d5b78ee..17fed93 100644
--- a/config/companion_tools.in
+++ b/config/companion_tools.in
@@ -19,6 +19,17 @@ config COMP_TOOLS
       If your system has older versions, we can build them for you,
       but you are strongly encouraged to update your system instead!
 
+if ! CONFIGURE_has_make
+
+config COMP_TOOLS_REQUIRE_make_3_81
+    bool
+    default y
+
+    select COMP_TOOLS
+    select COMP_TOOLS_make
+
+endif # ! CONFIGURE_has_make
+
 if COMP_TOOLS
 
 config COMP_TOOLS_make
diff --git a/configure.ac b/configure.ac
index 7da232c..f4ce583 100644
--- a/configure.ac
+++ b/configure.ac
@@ -221,6 +221,12 @@ AC_CACHE_CHECK([for GNU make >= 3.80], [ac_cv_path_MAKE],
 AC_SUBST([MAKE], [$ac_cv_path_MAKE])
 AC_PROG_MAKE_SET
 
+# Check for GNU make 3.81 exactly, otherwise
+# it will be built as a companion tool.
+AS_IF(
+      [test "$MAKE_ver" = "GNU Make 3.81"], [make="y"]
+     )
+ACX_SET_KCONFIG_OPTION([make])
 #----------------------------------------
 # Check for libtool >= 1.5.26
 AC_CACHE_VAL([ac_cv_path_LIBTOOL],
-- 
1.9.0

